### ISSUE TITLE
Import models from other location

### ISSUE THREAD

**Issue by Gank** [keystonejs/keystone#1246]
Hi,

I was trying to import a common model directory.
The following code works:

```
// Require keystone
var keystone = require('keystone');
keystone.init({});
keystone.import('models');
console.log(keystone.paths);
```

But if I change the models location it stops working, for example:
```
keystone.import('../models');
```

Error output:
```
module.js:340
    throw err;
    ^
Error: Cannot find module 'keystone'
  at Function.Module._resolveFilename (module.js:338:15)
  at Function.Module._load (module.js:280:25)
  at Module.require (module.js:364:17)
  at require (module.js:380:17)
  at Object.<anonymous> (/test/models/User.js:1:78)
  at Module._compile (module.js:456:26)
  at Object.Module._extensions..js (module.js:474:10)
  at Module.load (/test/app/node_modules/keystone/node_modules/cloudinary/node_modules/coffee-script/lib/coffee-script/register.js:45:36)
  at Function.Module._load (module.js:312:12)
  at Module.require (module.js:364:17)
  at require (module.js:380:17)
  at /test/node_modules/keystone/index.js:209:23
  at Array.forEach (native)
  at doImport (/test/app/node_modules/keystone/index.js:196:28)
  at [object Object].Keystone.import (/test/app/node_modules/keystone/index.js:218:9)
  at Object.<anonymous> (/test/test.js:6:16)
  at Module._compile (module.js:456:26)
  at Object.Module._extensions..js (module.js:474:10)
  at Module.load (module.js:356:32)
  at Function.Module._load (module.js:312:12)
  at Function.Module.runMain (module.js:497:10)
  at startup (node.js:119:16)
  at node.js:935:3

```

Any tips on how I can achieve this?

Thanks!

**Comment by molda** [comment]
change
```javascript
var keystone = require('keystone');
```
to point to location of keystone

so if you have your keystone app in /somedir/myapp

and models in /somedir/models

then you need 
```javascript
var keystone = require('./myapp/keystone');
```

**Comment by Gank** [comment]
Sorry, but that doesn't make much sense.

wouldn't it be, 

```
var keystone = require('../myapp/keystone'); 
```

Tried both ways but keeps giving another error.

```
fs.js:666
  return binding.readdir(pathModule._makeLong(path));
                 ^
Error: ENOENT, no such file or directory '/test/app/models'
    at Object.fs.readdirSync (fs.js:666:18)
    at doImport (/test/node_modules/keystone/index.js:202:6)
    at Keystone.import (/test/node_modules/keystone/index.js:224:9)
    at Object.<anonymous> (/test/keystone.js:32:16)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at Object.<anonymous> (/test/models/MyModel.js:1:78)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at /test/app/node_modules/keystone/index.js:215:23
    at Array.forEach (native)
    at doImport (/test/app/node_modules/keystone/index.js:202:28)
    at Keystone.import (/test/app/node_modules/keystone/index.js:224:9)
    at Object.exports.init (/test/app/controllers/database.js:19:18)
```

**Comment by morenoh149** [comment]
please read https://github.com/keystonejs/keystone/blob/40563de928892deb9f707a4498439663d154443d/index.js#L162 so
```javascript
keystone.import('../models');
```
would load models outside of the current project (doubt that's what you want).

And for the second question, you're trying to require keystone from a nonstandard location. Usually you'd do
```javascript
var keystone = require('keystone')
```
Which loads from `./node_modules/keystone/`. Please share more of your code.

**Comment by JohnnyEstilles** [comment]
@Gank unfortunately you can't do what you're trying to do, and it has nothing to do with KeystoneJS. It has to do with the NodeJS resolves required modules (see [Loading from node_modules Folders](https://nodejs.org/api/modules.html#modules_loading_from_node_modules_folders)).

Let me see if I can explain the situation.

Node searches for `required`modules (that are not native modules or start with `/`, `../`, or `./`) in:

>  "the parent directory of the current module, and adds /node_modules, and attempts to load the module from that location.If it is not found there, then it moves to the parent directory, and so on, until the root of the file system is reached."

Since Keystone models need to `require('keystone')`, these models must reside within any folder (including child folders) where the keystone module is installed.

So basically, you can:

```js
keystone.import('any/path/within/keystone/to/models'); 
```

but not:

```js
keystone.import('../any/path/outside/keystone/to/models');
```

**Comment by Gank** [comment]
Thank you @JohnnyEstilles, I'm going to use your suggestion.

**Comment by JohnnyEstilles** [comment]
@Gank ... you're very welcome! Let us know if you have any other questions.
